### ISSUE TITLE
[ Question ] CPU vs GPU DKL runs 

### ISSUE THREAD

**Issue by pythonpanda2** [cornellius-gp/gpytorch#1613]
I have set up an active learning runs for DKL  initialized by very low (3 - 5) data samples. The goal being to learn with minimum training samples relative to a standard NN. At each data iteration a new sample is labelled and added for retraining. The active learning  is run for ~ 1000 iteration. Everything runs fine in CPU. There are no error even at low data samples. The DKL class can be seen in the comment box below

Now when i run the same code  A100 GPU, there are two issues. The first issue is related to 

- Matrix not positive definite after repeatedly adding jitter up to 1.0e-04.

This was resolved by setting  noise_prior=gpytorch.priors.SmoothedBoxPrior(0.15, 1.5, sigma=0.5)  to the GaussianLikelihood

- After this fix there is no more errors related to torch.cholesky on GPU. Now in the new runs i get a very strange behavior.

With the CPU
 Iter 1/800 - Loss: 1.377
Iter 101/800 - Loss: 0.291
Iter 201/800 - Loss: -0.260
Iter 301/800 - Loss: -0.361
Iter 401/800 - Loss: -0.589
Iter 501/800 - Loss: -0.751
Iter 601/800 - Loss: -0.390
Iter 701/800 - Loss: -0.769
Iter 800/800 - Loss: -0.710

With the GPU
Iter 1/800 - Loss: 1.374
Iter 101/800 - Loss: 0.309
Iter 201/800 - Loss: -0.116
Iter 301/800 - Loss: -0.348
Iter 401/800 - Loss: -0.296
Iter 501/800 - Loss: -0.386
Iter 601/800 - Loss: -0.404
Iter 701/800 - Loss: -0.497
Iter 800/800 - Loss: 10.923
GPU loss becomes unbounded 

So the learning rate now is lr = 0.01 for both CPU and GPU.  Now when i reduce the GPU learning rate to lo = 0.001 (i.e. 1/10th of CPU LR) i get Stable behavior  as below

Iter 1/800 - Loss: 1.374
Iter 101/800 - Loss: 0.781
Iter 201/800 - Loss: 0.711
Iter 301/800 - Loss: 0.653
Iter 401/800 - Loss: 0.582
Iter 501/800 - Loss: 0.542
Iter 601/800 - Loss: 0.478
Iter 701/800 - Loss: 0.417
Iter 800/800 - Loss: 0.366

Please advice me If you have any idea why would there be such a deviation from CPU to GPU ? Does this mean GPU needs a lot more data ?

PyTorch==1.7.1+cu110 (Magma + CUDA 11.0)
GPyTorch==1.4.1
